File: back-end/Fretefy.Test.Domain/Dto/Regiao/NovaRegiaoDto.cs
using Fretefy.Test.Domain.Entities;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Runtime.CompilerServices;
using System.Text;

namespace Fretefy.Test.Domain.Dto
{
    public class NovaRegiaoDto : IValidatableObject
    {
        [Required]
        public string Nome { get; set; }
        [Required]
        public bool Status { get; set; }
        [Required]
        public List<Cidade> Cidades { get; set; }

        public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
        {
            if (Cidades == null || Cidades.Count == 0)
            {
                yield return new ValidationResult("A lista de cidades deve conter pelo menos uma cidade.", new[] { nameof(Cidades) });
            }
        }
    }
}

File: front-end/src/app/models/dto/cidade/regiao/nova-regiao-dto.ts
import { Cidade } from '../../../cidade'; 

export class NovaRegiaoDto {
  nome: string;
  status: boolean;
  cidades: Cidade[];

  constructor(nome: string, status: boolean, cidades: Cidade[]) {
    this.nome = nome;
    this.status = status;
    this.cidades = cidades;
  }
}